server/server.go

```diff
@@ -0,0 +1,328 @@
+// Package server implements the Bitswap protocol as a server
+package server
```

Comment:
We should figure out how to separate the server and client code in go-bitswap regardless of anything around auth. Bundling them together makes iterating and improving on either of them more difficult.